samples/java-android-app/src/main/java/com/amplitude/android/sample/MainApplication.java

```diff
@@ -84,4 +84,9 @@ public BaseEvent execute(@NonNull BaseEvent event) {
         event.getEventProperties().put("custom android event property", "test");
         return event;
     }
+
+    @Override
```

Comment:
@Mercy811 If we don't add this, would this reporting error? Would like to confirm how this interface change would affect users.

Reply:
Yes. This will cause an error 

> Class 'SamplePlugin' must either be declared abstract or implement abstract method 'teardown()' in 'Plugin'.

I did a research and find that it is due to differences in how Java and Kotlin handle default methods in interfaces

- In Kotlin, interfaces can have default implementations for methods, and any Kotlin class implementing such an interface is not required to override these methods with default implementations.
- However, when a Kotlin interface with default method implementations is used in Java, those default implementations are not automatically available. In Java, the class implementing the Kotlin interface is required to provide its own implementation for all methods, including those with default implementations in Kotlin.

So Kotlin app won't be affected, but plugins in Java app needs to add `teardown()`.

Reply:
@Mercy811 can we use default here to avoid interface errors?

Reply:
Yes! I updated in https://github.com/amplitude/Amplitude-Kotlin/pull/171/commits/f01adb912e3ba6be8a1fc12a7a67f726d4b098f1